components/top-bar/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react'
+import Headroom from 'react-headroom'
+import Link from 'next/link'
+import { StyledTopBar } from '@components/top-bar/styled'
+import { AppIcon } from '@components/logos'
+import { SearchBar } from '@components/search'
+import GetUpdatesModal from '@containers/modals/get-updates'
+import MenuIcon from 'mdi-react/MenuIcon'
+import CloseIcon from 'mdi-react/CloseIcon'
+import { Box } from 'blockstack-ui'
+import { Navigation } from '@components/navigation'
+import { generateLinkList } from '@components/navigation/helpers'
+import { primaryNavLinks, adminLinks } from '@common/constants'
+
+const handleBodyScroll = (on: boolean) =>
+  on
+    ? document.body.classList.remove('no-scroll')
+    : document.body.classList.add('no-scroll')
+
+const MenuToggle = ({ on }) => (
+  on ? <CloseIcon color="currentColor" /> : <MenuIcon color="currentColor" />
+)
+
+const TopBar = ({ isErrorPage, admin, wrap, ...props }) => {
+  const [menuOpen, setMenuOpen] = useState(false)
+
+  const NavLinks = () => (
+    <>
+      { admin && generateLinkList(adminLinks)({ isErrorPage }) }
+      { generateLinkList(primaryNavLinks)({ isErrorPage }) }
+    </>
+  )
```

Comment:
Curious what the pattern to follow is here. I _could_ create a totally separate component, but then I have to repeat the props. 
What's the best way to keep this clean and not repeat the contents?